### PR TITLE
Feature validation spec

### PR DESCRIPTION
**Pull request by GOlivierNation**
What is done in this PR:
    Add the following validations:
        For the User model:
            Name must not be blank.
            PostsCounter must be an integer greater than or equal to zero.
        For the Post model:
            Title must not be blank.
            Title must not exceed 250 characters.
            CommentsCounter must be an integer greater than or equal to zero.
            LikesCounter must be an integer greater than or equal to zero.
    Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Post.new(title: 'Hello', text: 'This is my first post', commentsCounter: 0, likesCounter: 0, user_id: user.id)
+  end
+
+  before { subject.save }
+
+  it 'title should be present' do
+    subject.title = nil
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/post_spec.rb]: - [ ] Kindly add tests for other [validations](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Title%20must%20not%20exceed%20250%20characters.,LikesCounter%20must%20be%20an%20integer%20greater%20than%20or%20equal%20to%20zero.) and for  methods.

[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.)

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  it 'should save the data' do
+    expect(subject).to be_valid
+  end
+
+  it 'name should be present' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/user_spec.rb]: - [ ] Kindly add tests for other [validations](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=PostsCounter%20must%20be%20an%20integer%20greater%20than%20or%20equal%20to%20zero.) and for  methods.

[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.)

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  it 'post_id should be a string' do
+    subject.post_id = '11123'
+    expect(subject).to_not be_valid
+  end
+
+  it 'user_id should be a string' do
+    subject.user_id = '1123'
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (ahmetbozaci) [inline spec/models/like_spec.rb]: - [ ] Kindly add test for `update_posts_counter` method.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @GOlivierNation  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by ahmetbozaci** [comment]
@GOlivierNation 

- [ ] Kİndly put unnecessary files to gitignore. As you can see many files in the pull request. It is hard to give a review with many files.

image

**Review by FrederickMih — APPROVED** [review]
Hi @GOlivierNation  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




## Highlights

- No Linter error :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**